api/books_test.go

```diff
@@ -41,6 +58,9 @@ func TestEndpoints(t *testing.T) {
 			Convey("And there AddBook function is not called", func() {
 				So(len(mockDataStore.AddBookCalls()), ShouldEqual, 0)
 			})
+			Convey("And the response says the request is empty", func() {
```

Comment:
Just an FYI you can also use `ShouldHaveLength` to assert the length of an array.

`So(mockDataStore.AddBookCalls(), ShouldHaveLength, 0)`

Functionally exactly the same, but IMO a little easier to read and the intention is clearer.

Reply:
I agree 🙂 - changing that now